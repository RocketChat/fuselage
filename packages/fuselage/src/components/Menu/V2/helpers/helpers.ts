import type { UsePositionOptions } from '@rocket.chat/fuselage-hooks';

type ReactAriaPlacement =
  | 'bottom'
  | 'bottom left'
  | 'bottom right'
  | 'bottom start'
  | 'bottom end'
  | 'top'
  | 'top left'
  | 'top right'
  | 'top start'
  | 'top end'
  | 'left'
  | 'left top'
  | 'left bottom'
  | 'start'
  | 'start top'
  | 'start bottom'
  | 'right'
  | 'right top'
  | 'right bottom'
  | 'end'
  | 'end top'
  | 'end bottom';

export const getPlacement = (
  placement: UsePositionOptions['placement'],
): ReactAriaPlacement => {
  // switch case for placement from usePosition placement to react-aria
  switch (placement) {
    case 'bottom':
      return 'bottom';

    case 'bottom-start':
      return 'bottom start';
    case 'bottom-end':
      return 'bottom end';
    case 'top':
      return 'top';

    case 'top-start':
      return 'top start';
    case 'top-end':
      return 'top end';
    case 'left':
      return 'left';
    case 'left-start':
      return 'left top';
    case 'left-end':
      return 'left bottom';
    case 'right':
      return 'right';
    case 'right-start':
      return 'right top';
    case 'right-end':
      return 'right bottom';
    default:
      return 'bottom start';
  }
};
