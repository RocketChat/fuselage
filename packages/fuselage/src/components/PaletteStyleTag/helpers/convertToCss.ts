export const convertToCss = (
  values: Record<string, string>,
  prefix: string,
  selector = ':root'
) =>
  `${selector} {\n${Object.entries(values)
    .map(([name, color]) => `${prefix}-${name}: ${color};`)
    .join('\n')}\n}`;
