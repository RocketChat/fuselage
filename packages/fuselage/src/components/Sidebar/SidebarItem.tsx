import type { IconName } from '@rocket.chat/icons';
import type { AllHTMLAttributes, ReactNode } from 'react';

import type { BoxProps } from '../Box';
import { Icon as FuselageIcon } from '../Icon';
import type { SidebarActionProps } from './SidebarActions';
import { SidebarAction, SidebarActions } from './SidebarActions';

/** @public */
export type SidebarItemProps = {
  selected?: boolean;
  highlighted?: boolean;
  clickable?: boolean;
  featured?: boolean;
  is?: BoxProps['is'];
  children?: ReactNode;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export function SidebarItem({
  selected,
  highlighted,
  clickable,
  featured,
  is: Tag = 'div',
  children,
  ...props
}: SidebarItemProps) {
  return (
    <Tag
      className={[
        'rc-box rcx-box--full rcx-sidebar-item',
        highlighted && 'rcx-sidebar-item--highlighted',
        (clickable || Tag === 'a') && 'rcx-sidebar-item--clickable',
        selected && 'rcx-sidebar-item--selected',
        featured && 'rcx-sidebar-item--featured',
      ]
        .filter(Boolean)
        .join(' ')}
      {...props}
    >
      <div
        className='rcx-box rcx-box--full rcx-sidebar-item__wrapper'
        children={children}
      />
    </Tag>
  );
}

type SidebarItemContainerProps = {
  children?: ReactNode;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export const SidebarItemContainer = (props: SidebarItemContainerProps) => (
  <div
    className='rc-box rcx-box--full rcx-sidebar-item__container'
    {...props}
  />
);

type SidebarItemMenuProps = {
  children?: ReactNode;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export const SidebarItemMenu = (props: SidebarItemMenuProps) => (
  <div
    className='rc-box rcx-box--full rcx-box--animated rcx-sidebar-item__menu-wrapper'
    {...props}
  />
);

type SidebarItemContentProps = {
  children?: ReactNode;
  className?: string;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export const SidebarItemContent = ({
  className = '',
  ...props
}: SidebarItemContentProps) => (
  <div
    className={`rc-box rcx-box--full rcx-sidebar-item__container rcx-sidebar-item__content ${className}`}
    {...props}
  />
);

type SidebarItemTitleProps = {
  children?: ReactNode;
  className?: string;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export const SidebarItemTitle = ({
  className = '',
  ...props
}: SidebarItemTitleProps) => (
  <div
    className={`rc-box rcx-box--full rcx-sidebar-item__title ${className}`}
    {...props}
  />
);

type SidebarItemTimeProps = {
  children?: ReactNode;
  className?: string;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export const SidebarItemTime = ({
  className,
  ...props
}: SidebarItemTimeProps) => (
  <div
    className={`rc-box rcx-box--full rcx-sidebar-item__time ${className}`}
    {...props}
  />
);

type SidebarItemBadgeProps = {
  children?: ReactNode;
  className?: string;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export const SidebarItemBadge = ({
  className,
  ...props
}: SidebarItemBadgeProps) => (
  <div
    className={`rc-box rcx-box--full rcx-sidebar-item__badge ${className}`}
    {...props}
  />
);

type SidebarItemSubtitleProps = {
  children?: ReactNode;
  className?: string;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export const SidebarItemSubtitle = ({
  className,
  ...props
}: SidebarItemSubtitleProps) => (
  <div
    className={`rc-box rcx-box--full rcx-sidebar-item__subtitle ${className}`}
    {...props}
  />
);

type SidebarItemWrapperProps = {
  children?: ReactNode;
  className?: string;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export const SidebarItemWrapper = ({
  className = '',
  ...props
}: SidebarItemWrapperProps) => (
  <div
    className={`rc-box rcx-box--full rcx-sidebar-item__wrapper ${className}`}
    {...props}
  />
);

type SidebarItemIconProps = {
  children?: ReactNode;
  className?: string;
  highlighted?: boolean;
  icon: IconName;
} & Omit<AllHTMLAttributes<HTMLElement>, 'name' | 'is'>;

/** @public */
export const SidebarItemIcon = ({
  highlighted,
  children,
  icon,
  className: _className,
  ...props
}: SidebarItemIconProps) => (
  <div
    className={[
      'rc-box rcx-box--full rcx-sidebar-item__icon',
      highlighted && 'rcx-sidebar-item__icon--highlighted',
    ]
      .filter(Boolean)
      .join(' ')}
    {...props}
  >
    {children || <FuselageIcon size='x16' name={icon} {...props} />}
  </div>
);

type SidebarItemAvatarProps = {
  children?: ReactNode;
} & AllHTMLAttributes<HTMLElement>;

/** @public */
export const SidebarItemAvatar = ({ ...props }: SidebarItemAvatarProps) => (
  <SidebarItemContainer>
    <div className='rc-box rcx-box--full rcx-sidebar-item__avatar' {...props} />
  </SidebarItemContainer>
);

/** @public */
export const SidebarItemActions = SidebarActions;

type SidebarItemActionProps = SidebarActionProps;

/** @public */
export const SidebarItemAction = (props: SidebarItemActionProps) => (
  <SidebarAction {...props} />
);

/** @public */
// eslint-disable-next-line @typescript-eslint/no-namespace
export namespace SidebarItem {
  export const Menu = SidebarItemMenu;
  export const Container = SidebarItemContainer;
  export const Content = SidebarItemContent;
  export const Title = SidebarItemTitle;
  export const Subtitle = SidebarItemSubtitle;
  export const Time = SidebarItemTime;
  export const Wrapper = SidebarItemWrapper;
  export const Icon = SidebarItemIcon;
  export const Avatar = SidebarItemAvatar;
  export const Actions = SidebarItemActions;
  export const Action = SidebarItemAction;
  export const Badge = SidebarItemBadge;
}

export default SidebarItem;
