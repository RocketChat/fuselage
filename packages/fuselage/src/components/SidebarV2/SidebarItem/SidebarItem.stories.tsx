import type { Meta, StoryFn } from '@storybook/react';

import { SidebarV2ListItem as SidebarListItem } from '../..';
import {
  GenericCondensedItem,
  GenericExtendedItem,
  GenericMediumItem,
  decorators,
} from '../helpers';

export default {
  title: 'Navigation/SidebarV2/Item',
  component: SidebarListItem,
  decorators,
} satisfies Meta<typeof SidebarListItem>;

export const Condensed: StoryFn<typeof SidebarListItem> = () => (
  <>
    {Array.from({ length: 8 }).map((_, i) => (
      <GenericCondensedItem key={i} i={i} />
    ))}
  </>
);

export const Medium: StoryFn<typeof SidebarListItem> = () => (
  <>
    {Array.from({ length: 8 }).map((_, i) => (
      <GenericMediumItem key={i} i={i} />
    ))}
  </>
);

export const Extended: StoryFn<typeof SidebarListItem> = () => (
  <>
    {Array.from({ length: 8 }).map((_, i) => (
      <GenericExtendedItem key={i} i={i} />
    ))}
  </>
);
