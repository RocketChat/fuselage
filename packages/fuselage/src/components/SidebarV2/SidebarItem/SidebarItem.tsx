import type { AllHTMLAttributes } from 'react';

export const SidebarItem = ({
  selected,
  level = 1,
  className,
  children,
  is: Tag = 'a',
  ...props
}: {
  selected?: boolean;
  level?: number;
  is?: React.ElementType;
} & AllHTMLAttributes<HTMLAnchorElement>) => (
  <Tag
    className={[
      'rcx-box rcx-box--full rcx-sidebar-v2-item',
      selected && 'rcx-sidebar-v2-item--selected',
      level && `rcx-sidebar-v2-item--level-${level}`,
      className,
    ]
      .filter(Boolean)
      .join(' ')}
    {...props}
  >
    {children}
  </Tag>
);
