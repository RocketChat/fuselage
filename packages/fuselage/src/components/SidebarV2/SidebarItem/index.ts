export { SidebarItem as SidebarV2Item } from './SidebarItem';
export { SidebarItemAction as SidebarV2ItemAction } from './SidebarItemAction';
export { SidebarItemAvatarWrapper as SidebarV2ItemAvatarWrapper } from './SidebarItemAvatarWrapper';
export { SidebarItemBadge as SidebarV2ItemBadge } from './SidebarItemBadge';
export { SidebarItemIcon as SidebarV2ItemIcon } from './SidebarItemIcon';
export { SidebarItemMenu as SidebarV2ItemMenu } from './SidebarItemMenu';
export { SidebarItemTitle as SidebarV2ItemTitle } from './SidebarItemTitle';
export { SidebarItemRow as SidebarV2ItemRow } from './SidebarItemRow';
export { SidebarItemCol as SidebarV2ItemCol } from './SidebarItemCol';
export { SidebarItemTimestamp as SidebarV2ItemTimestamp } from './SidebarItemTimestamp';
export { SidebarListItem as SidebarV2ListItem } from './SidebarListItem';
export { SidebarItemContent as SidebarV2ItemContent } from './SidebarItemContent';
export { SidebarItemStatusBullet as SidebarV2ItemStatusBullet } from './SidebarItemStatusBullet';
