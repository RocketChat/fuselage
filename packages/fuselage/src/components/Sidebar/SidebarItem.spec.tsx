import { testsFromStories } from '../../helpers/tests';
import * as stories from './SidebarItem.stories';

testsFromStories(stories);
