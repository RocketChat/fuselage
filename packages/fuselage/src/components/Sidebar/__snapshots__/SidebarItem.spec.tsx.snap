// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders Default without crashing 1`] = `
<body>
  <div>
    <div
      class="rc-box rcx-box--full rcx-sidebar-item"
    >
      <div
        class="rcx-box rcx-box--full rcx-sidebar-item__wrapper"
      >
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container"
        >
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__avatar"
          >
            <figure
              class="rcx-box rcx-box--full rcx-avatar rcx-avatar--x16"
            >
              <img
                alt=""
                class="rcx-avatar__element rcx-avatar__element--x16"
                src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCAAoACgDASIAAhEBAxEB/8QAGwAAAgIDAQAAAAAAAAAAAAAAAAcEBgIDBQj/xAAuEAACAQQAAwcEAQUAAAAAAAABAgMABAUREiExBhMUIkFRYQcWcYGhFTJSgpH/xAAYAQADAQEAAAAAAAAAAAAAAAACAwQBAP/EAB4RAAIBBQEBAQAAAAAAAAAAAAABAgMREiExE0HR/9oADAMBAAIRAxEAPwBuXuIkhBuMe5ib/AHQP49q4L3mLitryTLTSpOiHQI5k/HzXa/qbFOEudVTu1dumWvcTaNCZYZ7vU6g6LxqjOU/24dfs1Ouh9FnkMpd3Reeyx83hAxZZEhkdV9/MBrX71WGPvJcqrJBGveKATtuXXqNU0pu02bTHXD/AGvJAluyxxRd6F4x00o+NdKoVrjbzJdvVe1t5cVLc2ck8qjnohgpPtz2v7G6JtPQ2VJwjlcw+37mchpnK6GtIuv5NFWeTsLNPvxWTvpfjvOEfwKKzEVkSct2vscS/BIzSN0YRkeX81UpPqO8masJETu7OOccY4dswYFQeftv096XV5knuJGdm2T1+agvMXj8jEaHX905QihabvcbuS7X566mLWLwSY8PuRnk/u4eZ0deTl71Ef6hY+0yM88TzeNZY4luYwpVYyduOfrvhPTnr0pXSX9y5mCsyJMdyxxvwq599em+taItqCSNc90ChvZRUruUcT0JiO18Elpk7t8v41LWzacxkBSuvjQ/FFJayjDWrCTepAQ2vUH0oo/Jk3ovpwJJeVCP5CN+lFFaaMqy+nAyuChvrTI2kN9JAsi2ZOy4IBHMnkSCP+iqBexSWdxLazoUljJVlPUH2oorkV10pRc7b1zXb/hZOzuJvM86QWEXeELxOzHSIPcmiiiunVlF2RNTpRkrs//Z"
              />
            </figure>
          </div>
        </div>
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container rcx-sidebar-item__content "
        >
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__wrapper "
          >
            <div
              class="rc-box rcx-box--full rcx-sidebar-item__icon"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-lock rcx-icon rcx-css-1g87xs3"
              >
                
              </i>
            </div>
            <div
              class="rc-box rcx-box--full rcx-sidebar-item__title "
            >
              ddsfoasdhfiuashd ousadhfiuasdhfiuashd soduhfiuasdhfia ishdfiuashdfiuas iuashdfiausdh
            </div>
          </div>
        </div>
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container"
        >
          <div
            class="rcx-button-group rcx-button-group--align-start"
            role="group"
          >
            <button
              aria-label="accept"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary-success rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-phone rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
            <button
              aria-label="cancel"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary-danger rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-circle-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
            <button
              aria-label="delete"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary-info rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-trash rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
            <button
              aria-label="accept"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-phone rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </div>
      </div>
    </div>
    <a
      class="rc-box rcx-box--full rcx-sidebar-item rcx-sidebar-item--highlighted rcx-sidebar-item--clickable"
    >
      <div
        class="rcx-box rcx-box--full rcx-sidebar-item__wrapper"
      >
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container"
        >
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__avatar"
          >
            <figure
              class="rcx-box rcx-box--full rcx-avatar rcx-avatar--x28"
            >
              <img
                alt=""
                class="rcx-avatar__element rcx-avatar__element--x28"
                src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCAAoACgDASIAAhEBAxEB/8QAGwAAAgIDAQAAAAAAAAAAAAAAAAcEBgIDBQj/xAAuEAACAQQAAwcEAQUAAAAAAAABAgMABAUREiExBhMUIkFRYQcWcYGhFTJSgpH/xAAYAQADAQEAAAAAAAAAAAAAAAACAwQBAP/EAB4RAAIBBQEBAQAAAAAAAAAAAAABAgMREiExE0HR/9oADAMBAAIRAxEAPwBuXuIkhBuMe5ib/AHQP49q4L3mLitryTLTSpOiHQI5k/HzXa/qbFOEudVTu1dumWvcTaNCZYZ7vU6g6LxqjOU/24dfs1Ouh9FnkMpd3Reeyx83hAxZZEhkdV9/MBrX71WGPvJcqrJBGveKATtuXXqNU0pu02bTHXD/AGvJAluyxxRd6F4x00o+NdKoVrjbzJdvVe1t5cVLc2ck8qjnohgpPtz2v7G6JtPQ2VJwjlcw+37mchpnK6GtIuv5NFWeTsLNPvxWTvpfjvOEfwKKzEVkSct2vscS/BIzSN0YRkeX81UpPqO8masJETu7OOccY4dswYFQeftv096XV5knuJGdm2T1+agvMXj8jEaHX905QihabvcbuS7X566mLWLwSY8PuRnk/u4eZ0deTl71Ef6hY+0yM88TzeNZY4luYwpVYyduOfrvhPTnr0pXSX9y5mCsyJMdyxxvwq599em+taItqCSNc90ChvZRUruUcT0JiO18Elpk7t8v41LWzacxkBSuvjQ/FFJayjDWrCTepAQ2vUH0oo/Jk3ovpwJJeVCP5CN+lFFaaMqy+nAyuChvrTI2kN9JAsi2ZOy4IBHMnkSCP+iqBexSWdxLazoUljJVlPUH2oorkV10pRc7b1zXb/hZOzuJvM86QWEXeELxOzHSIPcmiiiunVlF2RNTpRkrs//Z"
              />
            </figure>
          </div>
        </div>
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container rcx-sidebar-item__content "
        >
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__wrapper "
          >
            <div
              class="rc-box rcx-box--full rcx-sidebar-item__icon"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-lock rcx-icon rcx-css-1g87xs3"
              >
                
              </i>
            </div>
            <div
              class="rc-box rcx-box--full rcx-sidebar-item__title "
            >
              highlighted
            </div>
          </div>
        </div>
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container"
        />
      </div>
    </a>
    <a
      class="rc-box rcx-box--full rcx-sidebar-item rcx-sidebar-item--clickable rcx-sidebar-item--selected"
    >
      <div
        class="rcx-box rcx-box--full rcx-sidebar-item__wrapper"
      >
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container"
        >
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__avatar"
          >
            <figure
              class="rcx-box rcx-box--full rcx-avatar rcx-avatar--x28"
            >
              <img
                alt=""
                class="rcx-avatar__element rcx-avatar__element--x28"
                src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCAAoACgDASIAAhEBAxEB/8QAGwAAAgIDAQAAAAAAAAAAAAAAAAcEBgIDBQj/xAAuEAACAQQAAwcEAQUAAAAAAAABAgMABAUREiExBhMUIkFRYQcWcYGhFTJSgpH/xAAYAQADAQEAAAAAAAAAAAAAAAACAwQBAP/EAB4RAAIBBQEBAQAAAAAAAAAAAAABAgMREiExE0HR/9oADAMBAAIRAxEAPwBuXuIkhBuMe5ib/AHQP49q4L3mLitryTLTSpOiHQI5k/HzXa/qbFOEudVTu1dumWvcTaNCZYZ7vU6g6LxqjOU/24dfs1Ouh9FnkMpd3Reeyx83hAxZZEhkdV9/MBrX71WGPvJcqrJBGveKATtuXXqNU0pu02bTHXD/AGvJAluyxxRd6F4x00o+NdKoVrjbzJdvVe1t5cVLc2ck8qjnohgpPtz2v7G6JtPQ2VJwjlcw+37mchpnK6GtIuv5NFWeTsLNPvxWTvpfjvOEfwKKzEVkSct2vscS/BIzSN0YRkeX81UpPqO8masJETu7OOccY4dswYFQeftv096XV5knuJGdm2T1+agvMXj8jEaHX905QihabvcbuS7X566mLWLwSY8PuRnk/u4eZ0deTl71Ef6hY+0yM88TzeNZY4luYwpVYyduOfrvhPTnr0pXSX9y5mCsyJMdyxxvwq599em+taItqCSNc90ChvZRUruUcT0JiO18Elpk7t8v41LWzacxkBSuvjQ/FFJayjDWrCTepAQ2vUH0oo/Jk3ovpwJJeVCP5CN+lFFaaMqy+nAyuChvrTI2kN9JAsi2ZOy4IBHMnkSCP+iqBexSWdxLazoUljJVlPUH2oorkV10pRc7b1zXb/hZOzuJvM86QWEXeELxOzHSIPcmiiiunVlF2RNTpRkrs//Z"
              />
            </figure>
          </div>
        </div>
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container rcx-sidebar-item__content "
        >
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__wrapper "
          >
            <div
              class="rc-box rcx-box--full rcx-sidebar-item__icon"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-lock rcx-icon rcx-css-1g87xs3"
              >
                
              </i>
            </div>
            <div
              class="rc-box rcx-box--full rcx-sidebar-item__title "
            >
              ddsfoasdhfiuashd ousadhfiuasdhfiuashd soduhfiuasdhfia ishdfiuashdfiuas iuashdfiausdh
            </div>
          </div>
        </div>
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container"
        >
          <div
            class="rcx-button-group rcx-button-group--align-start"
            role="group"
          >
            <button
              aria-label="accept"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary-success rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-phone rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
            <button
              aria-label="cancel"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary-danger rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-circle-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
            <button
              aria-label="delete"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary-info rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-trash rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
            <button
              aria-label="accept"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-phone rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </div>
      </div>
    </a>
    <a
      class="rc-box rcx-box--full rcx-sidebar-item rcx-sidebar-item--clickable"
    >
      <div
        class="rcx-box rcx-box--full rcx-sidebar-item__wrapper"
      >
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container"
        >
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__avatar"
          >
            <figure
              class="rcx-box rcx-box--full rcx-avatar rcx-avatar--x36"
            >
              <img
                alt=""
                class="rcx-avatar__element rcx-avatar__element--x36"
                src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCAAoACgDASIAAhEBAxEB/8QAGwAAAgIDAQAAAAAAAAAAAAAAAAcEBgIDBQj/xAAuEAACAQQAAwcEAQUAAAAAAAABAgMABAUREiExBhMUIkFRYQcWcYGhFTJSgpH/xAAYAQADAQEAAAAAAAAAAAAAAAACAwQBAP/EAB4RAAIBBQEBAQAAAAAAAAAAAAABAgMREiExE0HR/9oADAMBAAIRAxEAPwBuXuIkhBuMe5ib/AHQP49q4L3mLitryTLTSpOiHQI5k/HzXa/qbFOEudVTu1dumWvcTaNCZYZ7vU6g6LxqjOU/24dfs1Ouh9FnkMpd3Reeyx83hAxZZEhkdV9/MBrX71WGPvJcqrJBGveKATtuXXqNU0pu02bTHXD/AGvJAluyxxRd6F4x00o+NdKoVrjbzJdvVe1t5cVLc2ck8qjnohgpPtz2v7G6JtPQ2VJwjlcw+37mchpnK6GtIuv5NFWeTsLNPvxWTvpfjvOEfwKKzEVkSct2vscS/BIzSN0YRkeX81UpPqO8masJETu7OOccY4dswYFQeftv096XV5knuJGdm2T1+agvMXj8jEaHX905QihabvcbuS7X566mLWLwSY8PuRnk/u4eZ0deTl71Ef6hY+0yM88TzeNZY4luYwpVYyduOfrvhPTnr0pXSX9y5mCsyJMdyxxvwq599em+taItqCSNc90ChvZRUruUcT0JiO18Elpk7t8v41LWzacxkBSuvjQ/FFJayjDWrCTepAQ2vUH0oo/Jk3ovpwJJeVCP5CN+lFFaaMqy+nAyuChvrTI2kN9JAsi2ZOy4IBHMnkSCP+iqBexSWdxLazoUljJVlPUH2oorkV10pRc7b1zXb/hZOzuJvM86QWEXeELxOzHSIPcmiiiunVlF2RNTpRkrs//Z"
              />
            </figure>
          </div>
        </div>
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container rcx-sidebar-item__content "
        >
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__container rcx-sidebar-item__content "
          >
            <div
              class="rc-box rcx-box--full rcx-sidebar-item__wrapper "
            >
              <div
                class="rc-box rcx-box--full rcx-sidebar-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-icon--name-lock rcx-icon rcx-css-1g87xs3"
                >
                  
                </i>
              </div>
              <div
                class="rc-box rcx-box--full rcx-sidebar-item__title "
              >
                ddsfoasdhfiuashd ousadhfiuasdhfiuashd soduhfiuasdhfia ishdfiuashdfiuas iuashdfiausdh
              </div>
            </div>
          </div>
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__container rcx-sidebar-item__content "
          >
            <div
              class="rc-box rcx-box--full rcx-sidebar-item__wrapper "
            >
              <div
                class="rc-box rcx-box--full rcx-sidebar-item__subtitle undefined"
              >
                ddsfoasdhfiuashd ousadhfiuasdhfiuashd soduhfiuasdhfia ishdfiuashdfiuas iuashdfiausdh
              </div>
            </div>
          </div>
        </div>
        <div
          class="rc-box rcx-box--full rcx-sidebar-item__container"
        >
          <div
            class="rc-box rcx-box--full rcx-sidebar-item__container"
          >
            <div
              class="rcx-button-group rcx-button-group--align-start"
              role="group"
            >
              <button
                aria-label="accept"
                class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary-success rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
                type="button"
              >
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-icon--name-phone rcx-icon rcx-css-4pvxx3"
                >
                  
                </i>
              </button>
              <button
                aria-label="cancel"
                class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary-danger rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
                type="button"
              >
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-icon--name-circle-cross rcx-icon rcx-css-4pvxx3"
                >
                  
                </i>
              </button>
              <button
                aria-label="delete"
                class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary-info rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
                type="button"
              >
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-icon--name-trash rcx-icon rcx-css-4pvxx3"
                >
                  
                </i>
              </button>
              <button
                aria-label="accept"
                class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-secondary rcx-button--square rcx-button--icon rcx-button rcx-button-group__item"
                type="button"
              >
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-icon--name-phone rcx-icon rcx-css-4pvxx3"
                >
                  
                </i>
              </button>
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>
</body>
`;
